Gems/PhysX/Code/Source/EditorMeshColliderComponent.cpp

```diff
@@ -93,88 +93,117 @@ namespace PhysX
-        }
+            const Physics::ShapeConfiguration* shapeConfiguration = shapeConfig.second.get();
+            AZ_Assert(shapeConfiguration, "GetShapeTypesInsideAsset: Invalid shape-collider configuration pair");
 
-        // It's enough looking at the first shape as the rest would be the same type.
-        const Physics::ShapeConfiguration* shapeConfiguration = shapeConfigList[0].second.get();
-        AZ_Assert(shapeConfiguration, "PhysXMeshAssetShapeTypeName: Invalid shape-collider configuration pair");
-
-        switch (shapeConfiguration->GetShapeType())
-        {
-        case Physics::ShapeType::CookedMesh:
+            switch (shapeConfiguration->GetShapeType())
             {
-                const Physics::CookedMeshShapeConfiguration* cookedMeshShapeConfiguration =
-                    static_cast<const Physics::CookedMeshShapeConfiguration*>(shapeConfiguration);
-                switch (cookedMeshShapeConfiguration->GetMeshType())
+            case Physics::ShapeType::CookedMesh:
                 {
-                case Physics::CookedMeshShapeConfiguration::MeshType::Convex:
+                    const Physics::CookedMeshShapeConfiguration* cookedMeshShapeConfiguration =
+                        static_cast<const Physics::CookedMeshShapeConfiguration*>(shapeConfiguration);
+                    switch (cookedMeshShapeConfiguration->GetMeshType())
                     {
-                        return assetName + " (Convex)";
-                    }
-                case Physics::CookedMeshShapeConfiguration::MeshType::TriangleMesh:
-                    {
-                        return assetName + " (Triangle Mesh)";
-                    }
-                default:
-                    {
-                        AZ_Error(
-                            "EditorProxyAssetShapeConfig",
-                            false,
-                            "PhysXMeshAssetShapeTypeName: Unexpected MeshType %d",
-                            static_cast<AZ::u32>(cookedMeshShapeConfiguration->GetMeshType()));
-                        return assetName;
+                    case Physics::CookedMeshShapeConfiguration::MeshType::Convex:
+                        {
+                            shapeTypes.push_back(ShapeType::Convex);
+                        }
+                    case Physics::CookedMeshShapeConfiguration::MeshType::TriangleMesh:
+                        {
+                            shapeTypes.push_back(ShapeType::TriangleMesh);
+                        }
+                    default:
+                        {
+                            shapeTypes.push_back(ShapeType::Invalid);
+                        }
                     }
+                    break;
+                }
+            case Physics::ShapeType::Sphere:
+            case Physics::ShapeType::Box:
+            case Physics::ShapeType::Capsule:
+                {
+                    shapeTypes.push_back(ShapeType::Primitive);
+                }
```

Comment:
update: It looks like we're missing a `break` here, otherwise I believe `Invalid` will also be added 😬

Reply:
great catch! It used to be a return, so I missed to put the break when I changed the code. Thanks!

Reply:
Good to hear! 🙂